build.gradle

```diff
@@ -118,3 +139,25 @@ dependencies {
 
     androidTestUtil 'androidx.test:orchestrator:1.4.0'
 }
+
+
+
+// get the most recent git commit ID in long and short form
+class GitCommit {
```

Comment:
Why not use?
Same as the stuff above.

```
def getVersionCode = { ->
    try {
        def stdout = new ByteArrayOutputStream()
        exec {
            commandLine 'git', 'rev-parse', 'HEAD`'
            standardOutput = stdout
        }
        return stdout.toString().trim()
    } catch (ignored) {
        return null
    }
}
```

Reply:
and `git rev-parse --short HEAD`

we could also generalize the execute and parsing code.